plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -174,10 +174,16 @@
      * The OpenShift deploy timeout in seconds:
      * See this issue for background of why for end users on slow wifi on their laptops
      * DeploymentConfigs usually barf: https://github.com/openshift/origin/issues/10531
+     *
+     * Lets default to 2 hours which should be plenty of time on a slow connection to download
+     * all the images for a YAML/JSON manifest (which can contain many separate images)
      */
-    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds")
+    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds", defaultValue = "7200")
```

Comment:
Now I remember a bit better: I started a discussion at https://github.com/fabric8io/fabric8-maven-plugin/pull/794 (would have loved to discuss this at this time, but it didn't take off) and the issue we had, that, when a pod fails to start for some reason and the DC retries, it still didn't break the deployment and only after the three hours comes back. Such a large timeout like 3h can only be a patch on a larger issue. Also, this blocking operation does not provide any progress feedback (which it could as it is blocking), so one never knows when to press CTRL-C. Such a large timeout is really a black box.

I agree with you having this as a synchronous operation without progress meters in OpenShift is a PITA. But  a long blocking timeout doesn't solve this really (only for particular use cases and you can configure for this particular use cases instead of making this the default)

BTW, why does 'gofabric8 deploy' calls the fabric8-maven-plugin so that it needs a larger timeout ? Didn't get this part.

How often do you deploy 7-10 DCs with a single Maven build (where you don't have also build the images in the cluster) ? if it is a rare use case (e.g. for fabric8 only), you would need only to set the timeout on this monster deployment then.

@jimmidyson any opinions on a 2 hour deployment timeout ?